src/server/game/Entities/Unit/StatSystem.cpp

```diff
@@ -1060,26 +1052,78 @@ void Guardian::UpdateArmor()
 void Guardian::UpdateMaxHealth()
 {
     UnitMods unitMod = UNIT_MOD_HEALTH;
-    float stamina = GetStat(STAT_STAMINA) - GetCreateStat(STAT_STAMINA);
 
-    float multiplicator;
-    switch (GetEntry())
+    uint64 health = GetMaxHealth();
+    float value = 0.0f;
+    float pctFromOwnerHealth = 0.0f;
+
+    PetType petType = IsHunterPet() ? HUNTER_PET : SUMMON_PET;
+
+    switch (petType)
+    {
+    case SUMMON_PET:
     {
-        case ENTRY_IMP:         multiplicator = 8.4f;   break;
-        case ENTRY_VOIDWALKER:  multiplicator = 11.0f;  break;
-        case ENTRY_SUCCUBUS:    multiplicator = 9.1f;   break;
-        case ENTRY_FELHUNTER:   multiplicator = 9.5f;   break;
-        case ENTRY_FELGUARD:    multiplicator = 11.0f;  break;
-        case ENTRY_BLOODWORM:   multiplicator = 1.0f;   break;
-        default:                multiplicator = 10.0f;  break;
+        switch (GetEntry())
+        {
+        case ENTRY_BLOODWORM:
+            pctFromOwnerHealth = 15.f;
+            break;
+        case ENTRY_RISEN_SKULKER:
+            pctFromOwnerHealth = 20.f;
+            break;
+        case ENTRY_GHOUL:
+            pctFromOwnerHealth = 35.f;
+            break;
+        case ENTRY_XUEN:
+        case ENTRY_NIUZAO:
+        case ENTRY_CHI_JI:
+            pctFromOwnerHealth = 100.f;
+            break;
+        case ENTRY_FIRE_ELEMENTAL:
+            pctFromOwnerHealth = 75.f;
+            break;
+        case ENTRY_FELGUARD:
+        case ENTRY_VOIDWALKER:
+        case ENTRY_INFERNAL:
+        case ENTRY_WATER_ELEMENTAL:
+            pctFromOwnerHealth = 50.f;
+            break;
+        case ENTRY_FELHUNTER:
+        case ENTRY_SUCCUBUS:
+            pctFromOwnerHealth = 40.f;
+            break;
+        case ENTRY_IMP:
+            pctFromOwnerHealth = 30.f;
+            break;
+        default:
+            break;
+        }
+        break;
+    }
+    case HUNTER_PET:
+        pctFromOwnerHealth = 70.f;
+        break;
+    default:
+        break;
     }
 
-    float value = GetModifierValue(unitMod, BASE_VALUE) + GetCreateHealth();
+    if (m_Properties && m_Properties->ID == 3236)
+        pctFromOwnerHealth = 20.f;
+
+    if (pctFromOwnerHealth)
+        health = CalculatePct(m_owner->GetMaxHealth(), pctFromOwnerHealth);
+
+    // Pets do not have static base values
+    if (!(GetModifierValue(unitMod, BASE_VALUE) == health))
+        SetModifierValue(unitMod, BASE_VALUE, health);
+
+    value += GetModifierValue(unitMod, BASE_VALUE);
     value *= GetModifierValue(unitMod, BASE_PCT);
-    value += GetModifierValue(unitMod, TOTAL_VALUE) + stamina * multiplicator;
+    value += GetModifierValue(unitMod, TOTAL_VALUE);
     value *= GetModifierValue(unitMod, TOTAL_PCT);
 
-    SetMaxHealth((uint32)value);
+    SetCreateHealth((uint32)value); // Blizz is doing it... dont ask why
```

Comment:
Blizz is using uint32 here? what?

Reply:
Yes
Field is no float :/

Reply:
wait that doesn't make sense.. is CREATE_HEALTH the hp with which he got spawned ? If he got over 4.7m health, uint32 wouldn't fit for this.. MAX_HEALTH is 64 but not create health ?

Reply:
that was my thought too ^^
Need to check that